### ISSUE TITLE
[Slack] Messages in file threads get unthreaded

### ISSUE THREAD

**Issue by patcon** [42wim/matterbridge#608]
Came across this while testing https://github.com/42wim/matterbridge/pull/606

**Describe the bug**
When using a gateway from channel to channel on same slack team: posting a snippet (same as file) to one succeeds in relaying the file to the other channel. But when trying to reply to that file in a thread, the message gets unthreaded.

**To Reproduce**
Upload file or create snippet. Post a new message to file (not including "initial comment" from UI). check other channel and see unthreading.

**Expected behavior**
Expect new message to be threaded on other end of gateway.

**Screenshots/debug logs**
None.

**Environment (please complete the following information):**
 - OS: darwin. Using this PR branch https://github.com/42wim/matterbridge/pull/606

**Additional context**
File uploads have moved to using file threads: https://api.slack.com/changelog/2018-05-file-threads-soon-tread

This means that the comments are actually messages in the thread of the upload, and so are threaded via POSTing thread timestamps, which we call parentID. But the parentID that is generated in `b.uploadFile` for each file is never returned back to `BrMsgID` slice through `dest.Send`:

https://github.com/42wim/matterbridge/blob/8869e253cad84f504c5b90bfe9256ca8fb4e815c/bridge/slack/slack.go#L374-L404

https://github.com/42wim/matterbridge/blob/8869e253cad84f504c5b90bfe9256ca8fb4e815c/gateway/gateway.go#L330

Since there can potentially be multiple files to upload in a msg, what if we passed back `[]string` instead of `string`, and merged it in?

**Comment by 42wim** [comment]
Yes, this is an issue that should get fixed, not only for slack but also for other bridges where we could split up long lines in multiple lines. This will also return multiple mID.
Will require some work.

**Comment by 42wim** [comment]
The unthreading should be fixed on master btw, the issue that still exists is that files can't be deleted
